frame/dapps-staking/src/lib.rs

```diff
@@ -208,12 +368,39 @@ where
     }
 }
 
+/// Instruction on how to handle reward payout for stakers.
+/// In order to make staking more competitive, majority of stakers will want to
+/// automatically restake anything they earn.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum RewardHandling {
```

Comment:
We can change it to `RewardDestination`, sure!

For options, I believe `Stake` doesn't tell the entire story since we'll also deposit rewards into `Balance`.
I'd like to avoid people thinking that if they _re-stake_, the reward doesn't go immediately to their balance.

Would `Balance` and `StakeAndBalance` work?

Reply:
May be `StakeBalance` and `FreeBalance` in that case?

Reply:
Sure, sounds good.